addons/mail/models/mail_message.py

```diff
@@ -1250,3 +1250,49 @@ def _invalidate_documents(self, model=None, res_id=None):
 
     def _get_search_domain_share(self):
         return ['&', '&', ('is_internal', '=', False), ('subtype_id', '!=', False), ('subtype_id.internal', '=', False)]
+
+    def searchMessages(self, search_term, thread_model, thread_id, before=None):
+        domain = []
+        domain = expression.AND([
+            expression.OR([
+                [("body", "ilike", search_term)],
+                [("tracking_value_ids.field_desc", "ilike", search_term)],
+                [("tracking_value_ids.old_value_integer", "ilike", search_term)],
+                [("tracking_value_ids.old_value_float", "ilike", search_term)],
+                [("tracking_value_ids.old_value_monetary", "ilike", search_term)],
+                [("tracking_value_ids.old_value_char", "ilike", search_term)],
+                [("tracking_value_ids.old_value_text", "ilike", search_term)],
+                [("tracking_value_ids.old_value_datetime", "ilike", search_term)],
+                [("tracking_value_ids.new_value_integer", "ilike", search_term)],
+                [("tracking_value_ids.new_value_float", "ilike", search_term)],
+                [("tracking_value_ids.new_value_monetary", "ilike", search_term)],
+                [("tracking_value_ids.new_value_char", "ilike", search_term)],
+                [("tracking_value_ids.new_value_text", "ilike", search_term)],
+                [("tracking_value_ids.new_value_datetime", "ilike", search_term)],
+                [("subtype_id.description", "ilike", search_term)],
+                [("attachment_ids.name", "ilike", search_term)],
+                [("subject", "ilike", search_term)],
+            ]),
+        ])
+        if thread_id == "starred":
+            domain = expression.AND([domain, [("starred", "=", True)]])
+        elif thread_id == "inbox":
+            domain = expression.AND([domain, [("needaction", "=", True)]])
+        elif thread_id == "history":
+            domain = expression.AND([domain, [("needaction", "=", False)]])
+        else:
+            domain = expression.AND([
+                domain,
+                [("res_id", "=", int(thread_id))],
+                [("model", "=", thread_model)],
+                [("message_type", "!=", "user_notification")],
+            ])
+        messages = self.sudo()._message_fetch(domain, before=before)
+        messages = messages.filtered(self._filter_unauthorized_message_tracking_value)
```

Comment:
There is a mismatch between `_filter_unauthorized_message_tracking_value` and `_filter_unauthorized_message_by_tracking_value` aren't they supposed to be the same? If not `_filter_unauthorized_message_tracking_value` is undeclared

Reply:
You are right, I fixed it :)